### DISCUSSION TITLE
量算组件画出来的线颜色怎么改变？

### DISCUSSION THREAD

**Discussion by DuShuYuan** [zouyaoji/vue-cesium#162] [Q&A]
设置了  polylineMeasurementOpts 属性
 {
polylineOpts:{
width:2,
color:Cesium.Color.Blue,
colors:[Cesium.Color.Blue]
}
}

宽度变了，颜色没有变化

**Comment by zouyaoji** [comment]
Please refer to this structure (in the primitiveOpts property):
请参考这个结构（在primitiveOpts属性中）：

https://github.com/zouyaoji/vue-cesium/blob/dev/packages/composables/use-drawing/defaultOpts.ts#L198

**Reply by DuShuYuan** [reply]
额。。。不支持颜色修改啊。。。

**Reply by zouyaoji** [reply]
支持哈，不好意思刚才应该是说错了。在 primitiveOpts 中。

**Reply by DuShuYuan** [reply]
谢谢，看到了，传 primitiveOpts 属性是整体替换了原默认属性，这有点不太好。就比如我想只修改颜色，其他还想用默认效果，就必须把polylinePrimitiveOptsDefault属性全copy出来，再改颜色值才行。要是只传个颜色值进去就方便了

**Reply by zouyaoji** [reply]
我也觉得你说的方式好，但是我也没想好怎么解决这个问题。我试过深度拷贝，但是这样弄会导致prop丢失了响应性，不能动态更新了。

目前稍微好一点的办法是你像源码那样：自己导入我那个默认值，然后Object.assign修改自己要修改的内容。

**Reply by DuShuYuan** [reply]
这个属性不支持实时变化吗？只有初始配置一次，之后再改变颜色属性值就没有效果了

**Reply by zouyaoji** [reply]
我测试是可以的

**Reply by DuShuYuan** [reply]
我找到问题了
我是根据 polylinePrimitiveOptsDefault 创建了两个配置对象，颜色不同。然后根据用户操作，直接更换的primitiveOpts 对象，这样无效果；
但是只改color一个属性就有效果了。

**Reply by DuShuYuan** [reply]
创建两个polyline测算组件，设置不同的线颜色，结果是只有一个生效。。。。改一个，另一个也会变。。。。没招治了。。。

**Reply by zouyaoji** [reply]
我测试也没问题哈

**Reply by DuShuYuan** [reply]
我无语了。。。弄了个小demo  是可以的，到项目里就不行了。。。。

**Comment by ChauPha** [comment]
您好 您使用量算组件的时候可以默认收起吗

**Reply by zouyaoji** [reply]
通过 `mainFabOpts.modelValue`控制
